lib/matplotlib/backends/backend_qtagg.py

```diff
@@ -47,25 +48,20 @@ def paintEvent(self, event):
             right = left + width
             # create a buffer using the image bounding box
             bbox = Bbox([[left, bottom], [right, top]])
-            buf = memoryview(self.copy_from_bbox(bbox))
+            region = self.copy_from_bbox(bbox)
 
-            if QT_API == "PyQt6":
-                from PyQt6 import sip
-                ptr = int(sip.voidptr(buf))
-            else:
-                ptr = buf
+            # Now draw the region we copied with the painter.
+            r_width, r_height = region.get_bounds()[2:]
 
-            painter.eraseRect(rect)  # clear the widget canvas
```

Comment:
Note: I cannot see why we would need this call. Perhaps if the captured region has some alpha... ?

It has been there since https://github.com/matplotlib/matplotlib/blame/d7e050bf59fbed6a0e0cdac48899c7fcb5df534c/lib/matplotlib/backends/backend_qtagg.py#L56 at least. Further back it looks like it was in https://github.com/matplotlib/matplotlib/commit/feb16f1836787b0d0f75e89afa9ed0971d5c88c3. 

The MR at https://github.com/matplotlib/matplotlib/pull/13050 looks very relevant.

I have no problem restoring this line if there are concerns.

Reply:
I convinced myself to maintain that line, even though I don't know exactly what it would fix. I couldn't reproduce the original issue that the MR was solving.